### DISCUSSION TITLE
I am having a bit of a problem where the index is larger than the index

### DISCUSSION THREAD

**Discussion by TheGloriousMoron** [SebLague/Chess-Challenge#455] [Q&A]
public List<int> MoveSequence(Board futureBoard)
    {    
        for (int i = 0; i < random.Next(1, 5);)
        {
            Move[] futureMoves = futureBoard.GetLegalMoves();

            int randomMove = random.Next(futureMoves.Length);

            movePath.Add(randomMove);

            futureBoard.MakeMove(futureMoves[randomMove]);

        }
        return movePath;
    }

**Comment by ryanheath** [comment]
There may not be a future move, you should check the length of `futureMoves` to be greater than 0.

**Reply by TheGloriousMoron** [reply]
thanks I'll try that
